cmd/html/script.js

```diff
@@ -0,0 +1,200 @@
+document.addEventListener("DOMContentLoaded", () => {
+
+    // create nodes on svg based on getNodes() function
+    mapPodsToNodes().then(nodeData=>{
+        // create svg
+        const svg = d3.select("#visualization")
+        .append("svg")
+        .attr("width", window.innerWidth)
+        .attr("height", window.innerHeight);
+        
+        console.log(nodeData)
+        // create a simple root node
+        const root = d3.hierarchy(nodeData)
+        .sum(d => computeSize(d))
+        .sort();
+        
+        // print root node
+    console.log("Root node",root)
+
+    // create a treemap with nodeData
+    const treemap = d3.treemap()
+    // .title("kubernetes time machine")
+    // window.innerWidth, window.innerHeight
+    .size([600,600])
+        .paddingInner(10)
+        .paddingTop(2)
+        .paddingRight(2)
+        .paddingBottom(2)
+        .paddingLeft(2)
+        .round(2)
+        // .padding(0)
+        .tile(d3.treemapSquarify.ratio(1))(root)
+
+        console.log("Treemap",treemap)
+        // console.log(treemap.leaves())    
+        
+    const g = svg.append("g")
+    .attr("transform", "translate(" + treemap.leaves()[0].x0 + "," + treemap.leaves()[0].y0 + ")");
+    
+
+    
+
+    // add nodes
+    g.selectAll("a")
+    .data(treemap.descendants().slice(0))
+    .enter()
+    .append("a")
+    .attr("href", (d) => {
+      if (d.depth === 0) {
+        return "#"
+      }
+      return d.height === 1 ? "node/"+d.data.name : "pod/"+d.data.name
+      return d.data.name+"/"
+    
+    })
+
+    // add rectangles for each pod
+    g.selectAll("a")
+    .data(treemap.descendants().slice(0))
```

Comment:
This will get very unmaintainable very fast (and harder to test). Looking at it more closely I would highly recommend using a library.

Also consider that without types it hard to know what is going on

Reply:
This is already using a d3.js lib. I'm all out of ideas :joy: